lib/lure/vcs/hg.go

```diff
@@ -14,61 +17,63 @@ type HgRepo struct {
 	remotePath    string
 	trashBranch   string
 	defaultBranch string
+	authArgs      []string
 }
 
-func HgSanitizeBranchName(name string) string {
-	reg, _ := regexp.Compile("[^a-zA-Z0-9/_-]+")
-	safe := reg.ReplaceAllString(name, "_")
-	return safe
-}
-
-// HgClone clones repository locally
-func HgClone(auth Authentication, source string, to string, defaultBranch string, trashBranch string, basePath string) (HgRepo, error) {
-	var repo HgRepo
-
-	args := []string{"clone", source, to}
-
-	switch auth := auth.(type) {
-	case TokenAuth:
-		source = strings.Replace(source, "://", fmt.Sprintf("://x-token-auth:%s@", auth.Token), 1)
-		args = []string{"clone", source, to}
-	case UserPassAuth:
-		args = append([]string{
-			"--config", "auth.repo.prefix=*",
-			"--config", "auth.repo.username=" + auth.Username,
-			"--config", "auth.repo.password=" + auth.Password,
-		}, args...)
-	}
-
-	if _, err := Execute("", "hg", args...); err != nil {
-		return repo, err
-	}
-
+func NewHg(auth Authentication, source string, to string, defaultBranch string, trashBranch string, basePath string) (HgRepo, error) {
 	var workingPath strings.Builder
 	workingPath.WriteString(to)
 	workingPath.WriteString("/")
 	workingPath.WriteString(basePath)
 
-	repo = HgRepo{
+	repo := HgRepo{
 		workingPath:   workingPath.String(),
 		localPath:     to,
-		remotePath:    source,
+		remotePath:    auth.AuthenticateURL(source),
 		defaultBranch: defaultBranch,
 		trashBranch:   trashBranch,
 	}
 
-	switch auth := auth.(type) {
-	case UserPassAuth:
-		repo.SetUserPas(auth.Username, auth.Password)
-	}
+	//TODO : validate that it works
```

Comment:
Was it done? Can we log an issue with the content of this TODO instead of putting it in code? :P